File: indic-script-converter-python/tools/latn_iast_transliteration_verification/latn_iast_transcription_test.py
from __future__ import annotations

from indic_script_converter.latn_iast_transcription import (
    FinalAPolicy,
    IastPlainEnglishOptions,
    JnaPolicy,
    PlainEnglishRomanizationProfile,
    to_plain_english_from_iast,
)
from indic_script_converter.tools.example_latn_iast import transliteration_smoke_samples


def main() -> None:
    print('----------------------------------------------------------------')
    print(' 3. PLAIN ENGLISH TRANSLITERATOR SAMPLES (DEFAULT OPTIONS)')
    print('----------------------------------------------------------------')

    for source in transliteration_smoke_samples:
        print(f'"{source}" -> "{to_plain_english_from_iast(source)}"')

    _run_policy_tests()
    _run_profile_tests()


def _run_policy_tests() -> None:
    print('\n[Plain English Option: Keep Final "a" (scholarly)]')
    keep_a = IastPlainEnglishOptions(final_a=FinalAPolicy.KEEP)
    for source in ['Rāma', 'vrata', 'Kṛṣṇa', 'Lakṣmaṇa', 'yātrā']:
        print(f'  "{source}" -> "{to_plain_english_from_iast(source, options=keep_a)}"')

    print('\n[Plain English Option: Always Drop Final "a" (Hindi-style)]')
    drop_a = IastPlainEnglishOptions(final_a=FinalAPolicy.DROP)
    for source in ['Rāma', 'vrata', 'Kṛṣṇa', 'Lakṣmaṇa', 'yātrā']:
        print(f'  "{source}" -> "{to_plain_english_from_iast(source, options=drop_a)}"')

    print('\n[Plain English Option: jñ as "jna"]')
    jna = IastPlainEnglishOptions(jna=JnaPolicy.JNA)
    for source in ['jñāna', 'yajña']:
        print(f'  "{source}" -> "{to_plain_english_from_iast(source, options=jna)}"')


def _run_profile_tests() -> None:
    print('\n[Plain English Profile: extendedIndic]')
    extended = IastPlainEnglishOptions(
        profile=PlainEnglishRomanizationProfile.EXTENDED_INDIC,
    )
    for source in ['xaṇḍa', 'xaiva', 'qaum']:
        print(f'  "{source}" -> "{to_plain_english_from_iast(source, options=extended)}"')

    print('\n[Plain English Profile: Hunterian (explicitly lossy view)]')
    hunterian = IastPlainEnglishOptions(
        profile=PlainEnglishRomanizationProfile.HUNTERIAN,
    )
    for source in [
        'Rāma',
        'Kṛṣṇa',
        'Lakṣmaṇa',
        'laṛkā',
        'Rāmacandra',
        'Gorakhapura',
        'Sarasvatī',
        'Īśvara',
        'pañcāṅga',
        'duḥkha',
        'Devadatta',
        'Jaideva',
        'Kalyāṇapura',
        'Nārāyaṇapura',
        'Hariprasāda',
        'Kṛṣṇadāsa',
    ]:
        print(f'  "{source}" -> "{to_plain_english_from_iast(source, options=hunterian)}"')

    print('\n[Plain English Profile: strictIast blocks Hunterian-only rules]')
    strict = IastPlainEnglishOptions(
        profile=PlainEnglishRomanizationProfile.STRICT_IAST,
    )
    for source in ['Sarasvatī', 'Rāmacandra', 'duḥkha']:
        print(f'  "{source}" -> "{to_plain_english_from_iast(source, options=strict)}"')


if __name__ == '__main__':
    main()

File: dart/tool/latn_iast_transliteration_verification/latn_iast_transcription_test.dart
// ignore_for_file: avoid_print

import 'package:lipimala/latn_iast_transcription.dart';
import 'example_latn_iast.dart' as ex;

export '../../lib/latn_iast_transcription.dart';

void main() {
  print('----------------------------------------------------------------');
  print(' 3. PLAIN ENGLISH TRANSLITERATOR SAMPLES (DEFAULT OPTIONS)');
  print('----------------------------------------------------------------');

  for (final source in ex.transliterationSmokeSamples) {
    print('"$source" -> "${source.toPlainEnglishFromIast()}"');
  }

  _runPolicyTests();
  _runProfileTests();
}

void _runPolicyTests() {
  print('\n[Plain English Option: Keep Final "a" (scholarly)]');
  const keepA = IastPlainEnglishOptions(finalA: FinalAPolicy.keep);
  for (final source in ['Rāma', 'vrata', 'Kṛṣṇa', 'Lakṣmaṇa', 'yātrā']) {
    print('  "$source" -> "${source.toPlainEnglishFromIast(options: keepA)}"');
  }

  print('\n[Plain English Option: Always Drop Final "a" (Hindi-style)]');
  const dropA = IastPlainEnglishOptions(finalA: FinalAPolicy.drop);
  for (final source in ['Rāma', 'vrata', 'Kṛṣṇa', 'Lakṣmaṇa', 'yātrā']) {
    print('  "$source" -> "${source.toPlainEnglishFromIast(options: dropA)}"');
  }

  print('\n[Plain English Option: jñ as "jna"]');
  const jna = IastPlainEnglishOptions(jna: JnaPolicy.jna);
  for (final source in ['jñāna', 'yajña']) {
    print('  "$source" -> "${source.toPlainEnglishFromIast(options: jna)}"');
  }
}

void _runProfileTests() {
  print('\n[Plain English Profile: extendedIndic]');
  const extended = IastPlainEnglishOptions();
  for (final source in ['xaṇḍa', 'xaiva', 'qaum']) {
    print(
      '  "$source" -> "${source.toPlainEnglishFromIast()}"',
    );
  }

  print('\n[Plain English Profile: Hunterian (explicitly lossy view)]');
  const hunterian = IastPlainEnglishOptions(
    profile: PlainEnglishRomanizationProfile.hunterian,
  );
  for (final source in [
    'Rāma',
    'Kṛṣṇa',
    'Lakṣmaṇa',
    'laṛkā',
    'Rāmacandra',
    'Gorakhapura',
    'Sarasvatī',
    'Īśvara',
    'pañcāṅga',
    'duḥkha',
    'Devadatta',
    'Jaideva',
    'Kalyāṇapura',
    'Nārāyaṇapura',
    'Hariprasāda',
    'Kṛṣṇadāsa',
  ]) {
    print(
      '  "$source" -> "${source.toPlainEnglishFromIast(options: hunterian)}"',
    );
  }

  print('\n[Plain English Profile: strictIast blocks Hunterian-only rules]');
  const strict = IastPlainEnglishOptions(
    profile: PlainEnglishRomanizationProfile.strictIast,
  );
  for (final source in ['Sarasvatī', 'Rāmacandra', 'duḥkha']) {
    print('  "$source" -> "${source.toPlainEnglishFromIast(options: strict)}"');
  }
}
